src/solvers/petsc_nonlinear_solver.C

```diff
@@ -801,14 +891,25 @@ PetscNonlinearSolver<T>::solve (SparseMatrix<T> &  pre_in,  // System Preconditi
   Mat J;
   ierr = SNESGetJacobian(_snes,&J,PETSC_NULL,PETSC_NULL,PETSC_NULL);
   LIBMESH_CHKERR(ierr);
-  ierr = MatMFFDSetFunction(J, __libmesh_petsc_snes_mffd_interface, this);
+  ierr = MatMFFDSetFunction(J, libmesh_petsc_snes_mffd_interface, this);
   LIBMESH_CHKERR(ierr);
 #if !PETSC_RELEASE_LESS_THAN(3, 8, 4)
   // Resue the residual vector from SNES
   ierr = MatSNESMFSetReuseBase(J, PETSC_TRUE);
   LIBMESH_CHKERR(ierr);
 #endif
 
+  SNESLineSearch linesearch;
+  if (linesearch_object)
+  {
+    ierr = SNESGetLineSearch(_snes, &linesearch);
```

Comment:
@lindsayad is currently working on some backwards compatibility guards for these lines of code.